### PR TITLE
Attendance pagination

### PR DESCRIPTION
**Pull request by athullya-geekywolf**

### CODE REVIEW

File: src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                    DepartureTime = attendanceRecord?.DepartureTime,
+                    InsideDuration = attendanceRecord?.InsideDuration,
+                    OutsideDuration = attendanceRecord?.OutsideDuration,
+                    MissedPunch = attendanceRecord?.MissedPunch,
+                    AttendanceStatusId = statusId
+                });
+            }
+
+            int totalCount= attendanceHistory.Count;
+            int perPageCount = 15;
```
Comment (sanalsunnyGW) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQueryHandler.cs:121]: Accept this from FE

File: src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using MediatR;
+using WolfDen.Application.DTOs.Attendence;
+
+namespace WolfDen.Application.Requests.Queries.Attendence.AttendanceHistory
+{
+    public class AttendanceHistoryQuery:IRequest<List<WeeklySummaryDTO>>
```
Comment (sanalsunnyGW) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs:6]: Purpose of this query

File: src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    {
+        public int EmployeeId { get; set; }
+        public int Year { get; set; }
+        public int CurrentPage { get; set; }
+
```
Comment (sanalsunnyGW) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs:11]: Accept Pagesize and pageno

File: src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs (hunk 2 above)
Comment (athullya-geekywolf) [reply src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs:6]: To display the attendance history of employee (of chosen year)

File: src/services/WolfDen.Application/DTOs/Attendence/MonthlyReportDTO.cs [hunk 4]
```diff
@@ -6,13 +6,13 @@ public class MonthlyReportDTO
         public int Absent { get; set; }
         public int IncompleteShift { get; set; }
         public int Holiday  { get; set; }
-        public List<DateOnly> IncompleteShiftDays { get; set; }
-        public List<DateOnly> RestrictedHolidays { get; set; }
-        public List<DateOnly> NormalHolidays { get; set; }
-        public List<DateOnly> WFHDays { get; set; }
-        public List<DateOnly> LeaveDays { get; set; }
+        public string IncompleteShiftDays { get; set; }
+        public string RestrictedHolidays { get; set; }
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/DTOs/Attendence/MonthlyReportDTO.cs:10]: Why are these fields string?

File: src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQueryHandler.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+                            x.LeaveRequestStatusId == LeaveRequestStatus.Approved)
+                .ToListAsync(cancellationToken);
+
+            List<LeaveType> leaveTypes = await _context.LeaveType.ToListAsync(cancellationToken);
+
+            for (var currentDate = yearStart; currentDate < yearEnd; currentDate = currentDate.AddDays(1))
+            {
+                if (currentDate > today)
+                {
+                    break;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQueryHandler.cs:44]: Why break if no further execution is left?

**Review by GSARYAPUTHRAN — APPROVED** [review]

File: src/services/WolfDen.Application/DTOs/Attendence/MonthlyReportDTO.cs (hunk 4 above)
Comment (athullya-geekywolf) [reply src/services/WolfDen.Application/DTOs/Attendence/MonthlyReportDTO.cs:10]: List of dates is passed as string

File: src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs (hunk 2 above)
Comment (athullya-geekywolf) [reply src/services/WolfDen.Application/Requests/Queries/Attendence/AttendanceHistory/AttendanceHistoryQuery.cs]: modified with month choosing and dailystatus filter

File: src/services/WolfDen.Application/DTOs/Attendence/AttendanceHistoryDTO.cs [hunk 6]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
Comment (GSARYAPUTHRAN) [inline src/services/WolfDen.Application/DTOs/Attendence/AttendanceHistoryDTO.cs:1]: unused imports

**Review by GSARYAPUTHRAN — APPROVED** [review]

File: src/services/WolfDen.Application/DTOs/Attendence/AttendanceHistoryDTO.cs (hunk 6 above)
Comment (athullya-geekywolf) [reply src/services/WolfDen.Application/DTOs/Attendence/AttendanceHistoryDTO.cs]: removed

**Pull request merged**